### PR TITLE
Create projects

### PR DESCRIPTION
**Pull request by rkakodkar**
A bit of change in workflow. Add server and project details in .envs . Tested the docker on event service. Project investigators could not be directly added (not sure why), but investigators on relay are added as users on server on project creation.

First PR on this :)

### CODE REVIEW

File: xnat_admin_tools/create_new_projects.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import typer
+from dotenv import load_dotenv
+
+load_dotenv()
+
+app = typer.Typer()
+
+
+def get_user_details(connection: pyxnat.Interface):
+
```
Comment (mirestrepo) [inline xnat_admin_tools/create_new_projects.py:13]: Add a docstring with small description of this function

File: xnat_admin_tools/create_new_projects.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    for user in users:
+        firstname = connection.manage.users.firstname(user)
+        lastname = connection.manage.users.lastname(user)
+        users_details["{}, {}".format(lastname, firstname)] = user
+
+    return users_details
+
+
+def remove_empty(values_to_select, project_values):
+
```
Comment (mirestrepo) [inline xnat_admin_tools/create_new_projects.py:25]: Add a docstring with small description of this function

File: xnat_admin_tools/create_new_projects.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                values_to_insert["xnat:projectData/PI/lastname"] = value.split()[1]
+            else:
+                values_to_insert[insert_tag] = value
+
+    return values_to_insert
+
+
+@app.command()
+def create_new_projects(project_id: str):
+    # I used export env variables as well as hard coding the fields
```
Comment (mirestrepo) [inline xnat_admin_tools/create_new_projects.py:40]: Add a docstring with small description of this function

File: xnat_admin_tools/create_new_projects.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    try:
+        values_to_insert = remove_empty(values_to_select, project_values[0])
+        typer.echo("Creating project on server")
+        server_project = xserver_connection.select.project(project_id)
+
+        if not (server_project.exists()):
+            server_project.create(**values_to_insert)
+            typer.echo("Project {} created".format(project_id))
+
+            # Add users as owners
```
Comment (mirestrepo) [inline xnat_admin_tools/create_new_projects.py:96]: Maybe just expand this comment with the logic of what you are getting back from API?  is the `[0]` the  is lastname, firstname?

File: xnat_admin_tools/create_new_projects.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            for user in users[0]["project_invs"].split("<br/>"):
+                try:
+                    userid = user_details[user.strip()]
+                    server_project.add_user(userid, role="owner")
```
Comment (mirestrepo) [inline xnat_admin_tools/create_new_projects.py:100]: I thought we had success adding the PI via `xnat:projectData/PI/firstname` and `xnat:projectData/PI/lastname` that didn't work?

**Review by mirestrepo — CHANGES_REQUESTED** [review]
This looks great! Few minor changes in terms of comments and just a question about adding the PI.... Maybe investigators need ti be created before they can be added?

There is a url+ post to add investigators ... https://wiki.xnat.org/display/XAPI/Project+Investigators+API

**Comment by rkakodkar** [comment]
Adding the PI via firstname and lastname does work. Users and project inverstigators are separate from PI. A PI by default is included as a project investigator, and additional project investigators are provided as a result of `xnat:projectData/PROJECT_INVS` query in a string with <br\> seperator (there is a short description as comments of `create_new_projects.py`). However, adding project investigators via `xnat:projectData:PROJECT_INVS` doesn't work, nor does `xnat:projectData/PROJECT_INVS/firstname` and `xnat:projectData/PROJECT_INVS/lastname`.

File: xnat_admin_tools/create_new_projects.py (hunk 5 above)
Comment (rkakodkar) [reply xnat_admin_tools/create_new_projects.py:127]: It does work, PI is added via firstname/lastname. Check the function remove_empty. Users need to be added separately as owners independent of PI

**Comment by mirestrepo** [comment]
> Adding the PI via firstname and lastname does work. Users and project inverstigators are separate from PI. A PI by default is included as a project investigator, and additional project investigators are provided as a result of `xnat:projectData/PROJECT_INVS` query in a string with <br> seperator (there is a short description as comments of `create_new_projects.py`). However, adding project investigators via `xnat:projectData:PROJECT_INVS` doesn't work, nor does `xnat:projectData/PROJECT_INVS/firstname` and `xnat:projectData/PROJECT_INVS/lastname`.

We need to chat  - Isn's PI and Project Investigator the same? I'm confused...

**Comment by rkakodkar** [comment]
Also, it doesn't seem to be an issue of investigators not being present (checked by adding Mike as an investigator). I think it might be an issue with API call to project investigators.

**Review by mirestrepo — APPROVED** [review]

**Pull request merged**